docs/standard/serialization/binaryformatter-migration-guide/choose-a-serializer.md

```diff
@@ -0,0 +1,75 @@
+
+Serialization excludes non-public and readonly members unless specifically handled through attributes and constructors. System.Text.Json also supports [custom serialization and deserialization](../system-text-json/custom-contracts.md) for more control over how types are converted into JSON and vice versa. System.Text.Json does not support the `[Serializable]` attribute.
+
+[Migrate to System.Text.Json](./migrate-to-system-text-json.md).
+
+## XML using System.Runtime.Serialization.DataContractSerializer
+
+`DataContractSerializer` was introduced in .NET Framework 3.0 and is used to serialize and deserialize data sent in Windows Communication Foundation (WCF) messages. `DataContractSerializer` is an XML serializer that **fully supports the serialization programming model that was used by the `BinaryFormatter`**. It requires the known types to be specified up-front (but most .NET collections and primitive types are on a default allow-list and don't need to be specified). It's the serializer that requires the least amount of effort to migrate to.
+
+While `DataContractSerializer` carries those functional benefits when migrating from BinaryFormatter, it is not as performant as the other choices, nor does it use a cross-platform format.
+
+[Migrate to DataContractSerializer](./migrate-to-datacontractserializer.md).
+
+> [!NOTE]
+> Do not confuse `DataContractSerializer` with [`NetDataContractSerializer`](/dotnet/api/system.runtime.serialization.netdatacontractserializer). `NetDataContractSerializer` is also identified as a [dangerous serializer](../binaryformatter-security-guide#dangerous-alternatives).
+
+## Binary using MessagePack for C#
+
+MessagePack uses a highly efficient binary serialization format, resulting in smaller message sizes compared to JSON and XML. It's performant and ships with built-in LZ4 compression. The open-source [MessagePack for C#](https://github.com/MessagePack-CSharp/MessagePack-CSharp) library works best when all serializable public types and members are annotated with dedicated attributes. It does not serialize non-public types and members by default, but it can be customized. It supports read-only types and members by trying to select the best matching constructor. The constructor can be specified explicitly by using the `[SerializationConstructor]` attribute.
+
+[Migrate to MessagePack](./migrate-to-messagepack.md).
+
+## Binary using protobuf-net
+
+The protobuf-net library is a contract based serializer for .NET code that uses the binary "protocol buffers" serialization format. The API follows typical .NET patterns and is broadly comparable to `XmlSerializer` and `DataContractSerializer`. This popular library is also feature-rich and can handle non-public types and fields, but many scenarios do require applying attributes to members.
+
+[Migrate to protobuf-net](./migrate-to-protobuf-net.md).
```

Comment:
@mgravell Can you review this portion please? Feel free to push refinements directly to the PR/branch as you see fit.

Reply:
done with minor changes, thanks for the heads-up